core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -81,9 +86,17 @@ class ZooKeeperClient(connectString: String, sessionTimeoutMs: Int, connectionTi
       val responseQueue = new ArrayBlockingQueue[Req#Response](requests.size)
```

Comment:
Not related to this PR, but out of curiosity, why are we using a blocking queue here?

Reply:
You might be right. It doesn't look like a BlockingQueue is necessary because `CountDownLatch.await` makes the `happens-before` guarantee:
> Memory consistency effects: Until the count reaches zero, actions in a thread prior to calling countDown() happen-before actions following a successful return from a corresponding await() in another thread.
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CountDownLatch.html

Should this be addressed in this PR or later?

Reply:
Either way works for me.

Reply:
Cool. I'll do it in a different PR.